### PR TITLE
Boomers Ruin The Economy - Pt. 1: No More Free Materials

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
The tl;dr is this:
image
Things you print cost a fee for the materials that go directly to cargo. Finally, money will be useful after the 30 minute mark.

Listen, at first this might seem like a step in the wrong direction, but I have some logic behind it, so at least give it a try.

The full explanation is here, and I highly encourage you to read it: https://hackmd.io/@marksuckerberg/BJuxYU2ow#No-more-free-materials

## Why It's Good For The Game
See above link
tl;dr: Makes money useful, incentivizes mining more, potentially even increases person to person interactions. 

## Changelog
:cl:
add: A small fee has been imposed by the Supply department on the use of all materials, which is highly configurable.
tweak: Costs for materials are disabled on red alert, so emergencies will still be covered.
/:cl:

### CODE REVIEW

**Comment by triplezeta** [comment]
am I the only person who watches #github-logs like a hawk

**Comment by SynnGraffkin** [comment]
This does bring up a good question though, how does this work with stock parts when upgrading machines and shit? Are Science / Engie lads going to bankrupt themselves printing parts for stuff or will it pull from budget?

**Comment by triplezeta** [comment]
the answer is this:

the cost to print something inexpensive matwise like parts is negligible and a scientist/engineer paycheck can easily fit the bill

**Comment by SynnGraffkin** [comment]
Okay, but why should I pay money for parts that use only metal / glass using the roundstart materials in my department?

**Comment by Sparrowbird611** [comment]
Easy solution, we just get rid of said materials :)

**Comment by TheNeoGamer42** [comment]
does this apply to machines not linked to the ore silo? does this apply to ejecting mats?

**Comment by MarkSuckerberg** [comment]
Only applies to machines linked to a silo that has been configured to require a fee for mats, which the station's main silo is set to have by default. It does apply to ejecting mats so that people don't just build their own local lathes and use mats for free.

**Comment by MarkSuckerberg** [comment]
Also, I want to point out that all fees are dropped when on red alert, since it shouldn't be obstructing emergency response/repair.

**Comment by TheNeoGamer42** [comment]
does the fee automatically go to cargo or is there any way to set it to another account or something?

**Comment by MarkSuckerberg** [comment]
Fully configurable. You can change the linked ID, unlink the ID/budget card from it and have it be just like a pre-this PR silo, change the markup on prices of mats, and set whether or not people get compensation for putting materials into the silo (and at what threshold the silo's linked account should have at the least when determining payments), as well as I'm planning to add more configuration if the current configuration is used.

**Comment by SynnGraffkin** [comment]
So Traitor QM can just make everything expensive 4no'raisins.

**Comment by triplezeta** [comment]
Yes, just like how they could lock the ore silo from use before, or disconnect everyone from it. :moyai:

**Comment by SynnGraffkin** [comment]
I'm note entirely sure how I feel about this.

**Comment by Superyodeler** [comment]
This will probably need to be test merged to figure out what works and what needs to be changed. I think this will be a good start at making the economy more meaningful.

**Comment by Vorak2** [comment]
I see potential in this depending on how the rest is done

**Review by Superyodeler — APPROVED** [review]

**Comment by Superyodeler** [comment]
Added a fix for the drone NTOS bluescreens

**Review by Superyodeler — APPROVED** [review]
